### PR TITLE
[15 min fix] Show pinned article to logged out users

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

In #13807 we forgot that logged out users are served a server rendered feed that's different from that one served to authenticated users.

This PR adds the same behavior:

- pinned article first (unless pinned and featured are the same thing so it's going to be a combo pinned AND featured)
- featured article next 
- everything else after

## Related Tickets & Documents

#13807 

## QA Instructions, Screenshots, Recordings

1. Log in, pin an article, log out, check the article is still pinned
2. Log in, pin an article with the cover image, log out, check the article is still pinned
3. Make sure the pinned article doesn't appear twice in the feed

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

### CODE REVIEW

**Review by juliannatetreault — CHANGES_REQUESTED** [review]
So far, this is looking good in terms of being able to show pinned posts to logged out users and being able to pin and unpin posts from the post's show page (`_single_story.html.erb`)! 😄 

 That being said though, it seems as though pinned posts in the Admin Portal are no longer working as expected--I've uploaded a quick video for further context.

https://user-images.githubusercontent.com/32834804/125509452-708cb44c-da55-4433-8f22-61b717cf7f50.mov

Are you experiencing this as well, @rhymes?

File: app/views/articles/_single_story.html.erb [hunk 1]
```diff
@@ -48,6 +48,14 @@
           <a href="<%= story.path %>" class="crayons-story__tertiary fs-xs"><time datetime="<%= story.published_timestamp %>"><%= story.readable_publish_date %></time><span class="time-ago-indicator-initial-placeholder" data-seconds="<%= story.published_at_int %>"></span></a>
         </div>
       </div>
+
+      <% if defined?(pinned) && pinned == true %>
```
Comment (citizen428) [inline app/views/articles/_single_story.html.erb:52]: Under which circumstances would the local variable `pinned` not exist?

Comment (rhymes) [reply app/views/articles/_single_story.html.erb:52]: Love it! Didn't know about this injected variable :)

**Comment by rhymes** [comment]
@juliannatetreault I think that's technically a known behavior, you have to click "unpin post" in the button bar at the top to unpin a post :D Let me know if that works for you!

**Review by juliannatetreault — APPROVED** [review]
Tested this out locally and the pinned articles appeared as expected for logged out users! 🚀

**Comment by juliannatetreault** [comment]
It looks like the "license/cla" check is stuck for some reason. 🤔

**Review by citizen428 — APPROVED** [review]

**Pull request merged**